scripts/plot_dispatch.py

```diff
@@ -89,131 +278,42 @@ def reduce_labels(ax, simple_labels_dict):
-                df_demand=df_demand_time_filtered,
-                unit="W",
-                colors_odict=COLORS,
-            )
+            # Exchange region from bus_name with "ALL"
+            bus_name = "ALL_" + carrier
 
-            plt.grid()
-            plt.xlabel("Date (mm-dd)", loc="center", fontdict={"size": 17})
-            plt.ylabel("Power", loc="center", fontdict={"size": 17})
-            plt.xticks(fontsize=14)
-            plt.yticks(fontsize=14)
-            # format x-axis representing the dates
-            plt.gca().xaxis.set_major_formatter(mdates.DateFormatter("%Y-%m-%d"))
-            plt.gca().xaxis.set_major_locator(mdates.WeekdayLocator())
-
-            # Shrink current axis's height by 10% on the bottom
-            box = ax.get_position()
-            ax.set_position(
-                [box.x0, box.y0 + box.height * 0.15, box.width, box.height * 0.85]
+            # Aggregate bus data and demand
+            df_aggregated = dp.aggregate_timeseries(
+                df_stacked, columns_to_aggregate="region"
             )
-
-            # Simplify legend. As there is only one color per technology, there should
-            # be only one label per technology.
-            simple_labels_dict = {
-                "Battery": ["Battery out", "Battery in"],
-                "El. transmission external": ["El. import", "El. export"],
-                "El. transmission B-BB": [
-                    "El. transmission in",
-                    "El. transmission out",
-                ],
-                "El. shortage / curtailment": ["El. shortage", "Curtailment"],
-                "Heat cen. storage": ["Heat cen. storage out", "Heat cen. storage in"],
-                "Heat cen. mismatch": ["Heat cen. excess", "Heat cen. shortage"],
-                "Heat dec. storage": ["Heat dec. storage out", "Heat dec. storage in"],
-                "Heat dec. mismatch": ["Heat dec. excess", "Heat dec. shortage"],
-            }
-
-            handles, labels = reduce_labels(
-                ax=ax, simple_labels_dict=simple_labels_dict
+            df_demand_aggregated = dp.aggregate_timeseries(
+                df_demand_stacked, columns_to_aggregate="region"
             )
+            # Unstack aggregated bus data and demand
+            df_aggregated = dp.unstack_timeseries(df_aggregated)
+            df_demand_aggregated = dp.unstack_timeseries(df_demand_aggregated)
 
-            # Put a legend below current axis
+            plot_dispatch_data(df_aggregated, df_demand_aggregated)
 
-            ax.legend(
-                handles=handles,
-                labels=labels,
-                loc="upper center",
-                bbox_to_anchor=(0.5, -0.25),
-                fancybox=True,
-                ncol=4,
-                fontsize=14,
-            )
+            # Set df_stacked and df_demand_stacked to None as preparation for new cycle
+            df_stacked = None
+            df_demand_stacked = None
```

Comment:
Put this in the beginning of the loop and drop the statement before the loop.

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/242/commits/33dc7e64dd5ae60479a65c7fbe58ae89acd3b480.